moabb/evaluations/evaluations.py

```diff
@@ -35,6 +41,17 @@
 Vector = Union[list, tuple, np.ndarray]
 
 
+def objective(trial, X, y, clf, scorer, epochs):
+    learning_rate = trial.suggest_float("learning_rate", 1e-4, 1e-2, log=True)
+    weight_decay = trial.suggest_float("weight_decay", 1e-10, 1e-3, log=True)
+    drop_rate = trial.suggest_float("drop_rate", 0.0, 1.0)
+
+    model = create_deep_model(clf, learning_rate, weight_decay, drop_rate, epochs=epochs)
+    model.fit(X, y)
+
+    return scorer(model, X, y)
```

Comment:
here you are selecting the model based on the training loss while what you care about is the validation loss. so you should first make a train/val Split  and select the params based on the validation score

Reply:
It took me a while, but I managed to implement it. Super ugly, but it works!